mariadb_kernel/client_config.py

```diff
@@ -11,14 +11,30 @@ class ClientConfig:
     def __init__(self, log, name="mariadb_config.json"):
         self.log = log
         self.config_name = name
+
+        datadir = "/tmp/mariadb_kernel/datadir"
```

Comment:
"socket" and "mysqld.pid" should also go in /tmp/mariadb_kernel/ to minimize potential conflicts. We also need to add a os.makedirs("/tmp/mariadb_kernel/datadir", exist_ok=True) before running `myqsl_install_db` because AFAIK the script requires the directory tree to exist.

Reply:
Correction, I am silly again, `mysql_install_db` does run 'mkdir -p', so you only need to adjust the paths for socket file  and pid file.

Reply:
Fix in latest commit.